.circleci/config.yml

```diff
@@ -5,6 +5,214 @@
 
 version: 2.1
 
+commands:
+
+    restore_data_cache:
+        description: "Restore data cache"
+        steps:
+            - run:
+                name: Get cache key
+                command: |
+                    if [[ $(cat restore.txt) == "true" ]]; then
+                        date +%U > week_num;
+                    else
+                        echo 'missing' > week_num;
+                    fi
+            - restore_cache:
+                keys:
+                    - v1-small-cache-{{ checksum "week_num" }}-XxREeVGdS+mePo1KYK4gIXd9Qr4TY6MfIxq9Jzdo1qY=
```

Comment:
What would be a good key source? `requirements-dev.txt`?

Reply:
@bthirion these keys are computed using the datasets files, so we need to have them in order to compute the hash. At the `restore_cache` step, this isn't the case yet.
Note that, these values shouldn't change unless the datasets themselves change. If this happen, any new dataset will be cached with a different key (since the hash will be different), and on the next build the old cache key will be missed. This way, we will know that something changed in this dataset, such that we can investigate what caused this modification, and update the key if we decide to accept the change. (I hope this explanation is understandable...)
Maybe we don't need all this logic and simpler cache keys would be enough? WDYT?

Reply:
I see (your explanation is great), and I think that this is technically a good compromise.
I'm a bit worried about the long-term maintenance of these keys. I haven't checked: do other packages use this logic ?